### PR TITLE
Fixes #18962 - moved isc file parser into dhcp_common

### PR DESCRIPTION
**Pull request by dmitri-d**
These changes are required in order to support https://github.com/theforeman/smart_proxy_dhcp_remote_isc

### CODE REVIEW

**Comment by dmitri-d** [comment]
Any chance for a review? I'd like to get these changes in before 1.15 is branched.

**Comment by dmitri-d** [comment]
@domcleal: any chance you could take a look at this before this Monday?

File: modules/dhcp_common/isc/omapi_provider.rb [hunk 1]
```diff
@@ -1,10 +1,9 @@
 require 'time'
-require 'dhcp_isc/isc_common'
+require 'dhcp_common/server'
 
-module Proxy::DHCP::ISC
```
Comment (iNecas) [inline modules/dhcp_common/isc/omapi_provider.rb:4]: Keeping the `ISC` namespace would read a bit better to me

**Comment by iNecas** [comment]
The changes look reasonable. My only comment is regarding namespacing of the common ICS code, where I would expect the classes to be under 'ICS' submodule

File: modules/dhcp_common/isc/omapi_provider.rb (hunk 1 above)
Comment (dmitri-d) [reply modules/dhcp_common/isc/omapi_provider.rb:4]: ```::Proxy::DHCP::ISC``` namespace is reserved for the dhcp isc provider and can't be used for shared bits of code.

Comment (iNecas) [reply modules/dhcp_common/isc/omapi_provider.rb:4]: What about `ISCCommon` or something similar?

**Comment by dmitri-d** [comment]
- moved common dhcp isc code into CommonISC module.

**Comment by iNecas** [comment]
I have not run the code but :+1: code-wise + the tests look to cover the behaviour quite well to trust the change. I'm not core proxy contributor though, perhaps @lzap or @timogoebel would be willing into having a look?

**Review by iNecas — APPROVED** [review]

**Review by timogoebel — APPROVED** [review]
Code looks good to me as well. I gave this a quick spin. Works well.

**Pull request merged**